src/main/java/com/cburch/logisim/fpga/download/VivadoDownload.java

```diff
@@ -243,48 +244,33 @@ public boolean CreateDownloadScripts() {
     return FileWriter.WriteContents(loadBitstreamFile, contents);
   }
 
-  private ArrayList<String> GetPinLocStrings() {
-    var contents = new ArrayList<String>();
-    for (var key : MapInfo.getMappableResources().keySet()) {
-      var map = MapInfo.getMappableResources().get(key);
+  private ArrayList<String> getPinLocStrings() {
+    final var contents = new LineBuffer();
+    for (final var key : MapInfo.getMappableResources().keySet()) {
+      final var map = MapInfo.getMappableResources().get(key);
       for (var i = 0; i < map.getNrOfPins(); i++) {
         if (map.isMapped(i) && !map.IsOpenMapped(i) && !map.IsConstantMapped(i) && !map.isInternalMapped(i)) {
           final var netName = (map.isExternalInverted(i) ? "n_" : "") + map.getHdlString(i);
-          contents.add(
-              "set_property PACKAGE_PIN "
-                  + map.getPinLocation(i)
-                  + " [get_ports {"
-                  + netName
-                  + "}]");
+          contents.add("set_property PACKAGE_PIN %s [get_ports {%s}]", map.getPinLocation(i), netName);
           final var info = map.getFpgaInfo(i);
           if (info != null) {
-            if (info.GetIOStandard() != IoStandards.Unknown
-                && info.GetIOStandard() != IoStandards.DefaulStandard) {
-              contents.add(
-                  "    set_property IOSTANDARD "
-                      + IoStandards.GetConstraintedIoStandard(info.GetIOStandard())
-                      + " [get_ports {"
-                      + netName
-                      + "}]");
+            final var ioStandard = info.GetIOStandard();
+            // FIXME: this code looks like copy&paste&paste issue?
+            if (ioStandard != IoStandards.Unknown && ioStandard != IoStandards.DefaulStandard) {
```

Comment:
This looks to me also as copy-paste issue, as this part I never touched I would take out one of the two copies

Reply:
Removed.

Reply:
Having twice the pins mapped is no issue, but it is redundant.